### PR TITLE
feat(jtk): shared presentation primitives for table, block, and id modes

### LINKED ISSUE #232: jtk: add shared presentation primitives for table, block, and id modes
Parent: #230

Part of the JTK command surface area rethink tracked in #230.

## Acceptance criteria
- Shared render paths exist for table mode, single-entity block mode, block-mode lists, and identifier-only mode
- Empty collection behavior follows the stdout/stderr contract from the spec
- Pagination continuation lines render in the correct stream for default vs `--id`
- Commands can adopt the new primitives without duplicating output policy

### PR DESCRIPTION
**Pull request by rianjs**
Closes #232.

## Summary

- Adds minimal shared presentation primitives (`Emit`, `EmitIDs`, `EmitIDsWithPagination`) so jtk commands can delegate output policy rather than re-invent the render + split-streams boilerplate.
- Renderer change (`shared/present`): consecutive stdout-bound `DetailSection`s render with a blank line between them, enabling block-mode lists (e.g. `comments list --fulltext`) without a new `Section` type.
- Pagination continuation line routes to **stdout** (inline with data rows), matching the #230 spec examples and making single-stream agent parsing straightforward. Every pagination call site — the two `IssuePresenter.PresentListWithPagination`/`CommentPresenter.PresentList*WithPagination` presenters and the `EmitIDsWithPagination` direct-write path — funnels through `paginationMessageSection()` + `appendPaginationHint()` in `tools/jtk/internal/present/emit.go`, so wording, kind, and stream cannot drift.
- Adopts the primitives in three flagship commands proving each mode:
  - `issues get` — single-entity block
  - `issues list` — table + pagination + `--id`
  - `comments list` — block-mode list (`--fulltext`) + pagination + `--id`
- `comments list` text path previously discarded `hasMore`; it now renders the continuation line like `issues list`.

Remaining ~25 jtk commands continue to work via their existing paths; per-command migration to `Emit`/`EmitIDs` is tracked under #230 follow-ups.

## Behavior changes

- **Breaking (minor):** `jtk <list> 2>/dev/null` no longer suppresses the pagination hint — it now lives on stdout with the data rows.
- `--id` on list commands emits one identifier per line, with a continuation line appended on stdout when more pages exist (matches default-mode policy).
- Empty result sets continue to go to stdout via the existing `PresentEmpty()` helpers (behavior unchanged; now covered by command-level tests to lock the contract).

## Test plan

- [x] `make build` — clean
- [x] `make test` — full suite passes (`shared/present`: 25; `tools/jtk/internal/present`: 44; `tools/jtk/internal/cmd/issues`: 68; `tools/jtk/internal/cmd/comments`: 20 top-level funcs)
- [x] `make lint` — no new findings (8 pre-existing gosec issues in cfl/shared untouched)
- [ ] Manual smoke against a live instance:
  - `jtk issues get MON-... --id` → single key
  - `jtk issues list -p MON` → table with pagination hint on stdout when `| wc -l` includes the hint
  - `jtk issues list -p MON --id` → keys one per line + continuation line when hasMore
  - `jtk comments list MON-... --fulltext` → one block per comment, blank line between
  - `jtk comments list MON-... --id` → comment IDs one per line

## New / changed test coverage

- `shared/present/render_test.go`: 5 cases locking the consecutive-DetailSection separator rule
- `tools/jtk/internal/present/emit_test.go`: `Emit` / `EmitIDs` / `EmitIDsWithPagination` behavior including empty-and-hasMore edge, plus `TestPaginationMessageSection_Canonical` and `TestAppendPaginationHint` locking the centralized pagination builder/mutator
- `tools/jtk/internal/present/issue_test.go`: pagination hint now asserted on stdout (policy flip lock-in)
- `tools/jtk/internal/cmd/issues/list_test.go` (new): pagination + `--id` + empty behavior for `issues list`
- `tools/jtk/internal/cmd/comments/comments_test.go`: block spacing, pagination-on-stdout, `--id` per-line, `--id` + continuation, empty + default, empty + `--id`, `commentsHasMore` table test

### CODE REVIEW

**Comment by rianjs** [comment]
**TDD Coverage Review — PR #250**

Overall the coverage is thorough and the test strategy is sound. I found one genuine gap and one minor observation.

---

**Gap: `comments list --id` with empty result set — no command-level test**

In `comments.go`, the `EmitIDOnly()` guard (line 74) fires *before* the `len == 0` guard (line 82), so an empty `--id` call goes through `EmitIDsWithPagination(opts, []string{}, false)` — which correctly emits nothing. This is indirectly proven by `TestEmitIDsWithPagination_EmptyAndNoMore` in `emit_test.go`, but there is no end-to-end `runList` test that exercises this branch.

Compare: `issues/list_test.go` has `TestRunList_EmptyWithIDOnly_EmitsNothing` (line 125). The comments package has no equivalent.

Proposed test: `TestRunList_EmptyWithIDOnly_EmitsNothing` — set up `commentsServerWithTotal(nil, 0)`, set `opts.IDOnly = true`, call `runList`, assert both stdout and stderr are empty. This would lock the "empty + --id → silent" contract at the command boundary, preventing a future refactor from accidentally routing to `PresentEmpty` instead of the ID path.

---

**Observation: `PresentListWithPagination(hasMore=false)` path not exercised at command level for comments**

`TestRunList_FullTextBlockSpacing` and `TestRunList_FullTextPaginationOnStdout` cover the `noTruncate=true` path. The `noTruncate=false` path (truncated list, no pagination) is not tested at command level for comments. This is low risk — `PresentListWithPagination` is a thin wrapper with its own unit tests — but worth noting if the contract matters.

---

**What's covered well**

- `commentsHasMore` table test hits all five cases including the Total=0 heuristic and the mid-page boundary.
- `Emit` / `EmitIDs` / `EmitIDsWithPagination` unit tests cover empty-and-hasMore, no-more, and nil-slice edge cases.
- `MutationPresenter.Pagination()` correctly asserts StreamStdout, contrasted with `Advisory()` asserting StreamStderr.
- The consecutive-DetailSection separator has five targeted render tests covering all non-trivial combinations.
- `issues list` command tests cover all four quadrants: default+pagination, --id+last-page, --id+more-pages, empty+default, empty+--id.

**Comment by rianjs** [comment]
Added `TestRunList_EmptyWithIDOnly_EmitsNothing` in `comments_test.go` (commit 9fd0893) — the one command-level gap you flagged. The behavior was already covered indirectly by `emit_test.go`, but the `runList` contract is now locked explicitly, parallel to the same test in `issues/list_test.go`.

**Comment by rianjs** [comment]
Addressed in 75eea54: every pagination call site now funnels through a single `paginationMessageSection()` builder and `appendPaginationHint(sections, hasMore)` mutator. `issue.PresentListWithPagination`, both comment-list variants, and `EmitIDsWithPagination` now share construction so wording, kind, and stream cannot drift. Also dropped the dead `PresentPaginationHint` and `MutationPresenter.Pagination` helpers that the previous commit left in place. Tests added: `TestPaginationMessageSection_Canonical`, `TestAppendPaginationHint`.

File: shared/present/render.go [hunk 1]
```diff
@@ -8,18 +8,27 @@ import (
 
 // Render converts an OutputModel to RenderedOutput with proper stream routing.
 // This is a pure function with no side effects.
+//
```
Comment (monit-reviewer) [inline shared/present/render.go:11]: 🟡 **Medium** (harness-engineering:architecture-reviewer): The consecutive-DetailSection blank-line separator is in shared/present, which is used by both jtk and cfl. This PR is scoped feat(jtk) but the render behavior change is global. No cfl tests are added to guard against silent regressions in cfl commands that emit consecutive DetailSections (e.g., page detail views). If cfl produces blank-line artifacts post-merge, there is no test contract to catch it.

File: tools/jtk/internal/cmd/comments/comments.go [hunk 2]
```diff
@@ -69,39 +69,42 @@ func runList(ctx context.Context, opts *root.Options, issueKey string, maxResult
 		return err
 	}
 
+	hasMore := commentsHasMore(result.Total, result.StartAt, len(result.Comments), maxResults)
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/comments/comments.go:72]: 🔵 **Low** (security:code-auditor): commentsHasMore produces a false positive when the final page is exactly maxResults in size and Total==0: the heuristic branch evaluates `got == maxResults` as true even when no further pages exist. After this PR this hint surfaces on stdout in both text and --id modes, making the false positive user-visible. Additionally, the degenerate case commentsHasMore(0, 0, 0, 0) returns true because `0 == 0`; a `got == 0 { return false }` guard would eliminate this edge.

File: tools/jtk/internal/cmd/issues/list.go [hunk 3]
```diff
@@ -107,26 +106,26 @@ func runList(ctx context.Context, opts *root.Options, project, sprint string, ma
...
 
+	hasMore := !result.Pagination.IsLast
+
+	if opts.EmitIDOnly() {
+		ids := make([]string, len(result.Issues))
+		for i, issue := range result.Issues {
+			ids[i] = issue.Key
+		}
+		return jtkpresent.EmitIDsWithPagination(opts, ids, hasMore)
+	}
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/issues/list.go:118]: 🔵 **Low** (harness-engineering:knowledge-reviewer): When len(result.Issues)==0 the text path calls Emit(PresentEmpty()), which carries no pagination context. If the API returns an empty page with hasMore=true, the continuation hint is silently dropped in text mode. The --id path avoids this because EmitIDsWithPagination is called first, but the text path does not apply the same appendPaginationHint logic.

File: tools/jtk/internal/cmd/comments/comments.go [hunk 4]
```diff
@@ -69,39 +69,42 @@ func runList(ctx context.Context, opts *root.Options, issueKey string, maxResult
...
+
+	if opts.EmitIDOnly() {
+		ids := make([]string, len(result.Comments))
+		for i, c := range result.Comments {
+			ids[i] = c.ID
+		}
+		return jtkpresent.EmitIDsWithPagination(opts, ids, hasMore)
+	}
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/comments/comments.go:80]: 🔵 **Low** (harness-engineering:knowledge-reviewer): Same empty-result + hasMore asymmetry as issues/list.go: when len(result.Comments)==0 the code returns PresentEmpty() via Emit, discarding the already-computed hasMore value. A mid-stream empty page in text mode silently drops the continuation hint.

File: tools/jtk/internal/present/emit.go [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+	_, _ = fmt.Fprint(opts.Stdout, out.Stdout)
+	_, _ = fmt.Fprint(opts.Stderr, out.Stderr)
+	return nil
+}
+
+// EmitIDs writes one identifier per line to opts.Stdout. Empty input emits
+// nothing. Matches `kubectl get -o name` / `ls -1` semantics.
+func EmitIDs(opts *root.Options, ids ...string) error {
+	for _, id := range ids {
+		_, _ = fmt.Fprintln(opts.Stdout, id)
```
Comment (monit-reviewer) [inline tools/jtk/internal/present/emit.go:56]: 🔵 **Low** (harness-engineering:enforcement-reviewer): EmitIDs is variadic (ids ...string) while EmitIDsWithPagination takes a plain slice (ids []string). These are conceptually the same operation with asymmetric signatures. Call sites holding a slice must choose the pagination variant regardless of whether pagination is needed, and future migration of the remaining ~25 commands will encounter this inconsistency. Aligning both to the same convention would reduce friction.

File: tools/jtk/internal/present/comment.go [hunk 6]
```diff
@@ -53,7 +54,10 @@ func (CommentPresenter) PresentListFull(comments []api.Comment) *present.OutputM
 		}
```
Comment (monit-reviewer) [inline tools/jtk/internal/present/comment.go:54]: 🔵 **Low** (harness-engineering:enforcement-reviewer): PresentListFull trims trailing newlines from the ADF body so the new consecutive-DetailSection separator produces exactly one blank line. PresentList (the non-fulltext/truncated path) is not visible in this diff. If PresentList also emits DetailSections with trailing newlines, those blocks will produce double blank lines between them under the new separator rule. Verify PresentList is table-mode only or apply the same trim.

File: shared/present/render_test.go [hunk 7]
```diff
@@ -386,6 +386,115 @@ func TestRender_MessageSection_Error(t *testing.T) {
...
+			&TableSection{
+				Headers: []string{"K", "V"},
+				Rows:    []Row{{Cells: []string{"a", "b"}}},
+			},
+		},
+	}
+
+	out := Render(model, StyleAgent)
+	want := "ID: 1\nK | V\na | b\n"
+	if out.Stdout != want {
```
Comment (monit-reviewer) [inline shared/present/render_test.go:435]: 🔵 **Low** (harness-engineering:architecture-reviewer): TestRender_DetailThenStderrMessage_NoStdoutSeparator has an inverted name: the test asserts wantStdout = "ID: 1\n\nID: 2\n", which verifies that a separator IS present between consecutive DetailSections. The name says 'NoStdoutSeparator' but the assertion confirms the opposite, which will mislead future readers.

**Review by monit-reviewer — DISMISSED** [review]
## Automated PR Review



**Reviewed commit:** `75eea54`

### Summary

| Reviewer | Findings |
|----------|----------|
| harness-engineering:architecture-reviewer | 2 |
| harness-engineering:enforcement-reviewer | 2 |
| harness-engineering:knowledge-reviewer | 2 |
| security:code-auditor | 1 |

<details>
<summary><strong>harness-engineering:architecture-reviewer (2 findings)</strong></summary>

**⚠️ Should Fix** - `shared/present/render.go:11`
> The consecutive-DetailSection blank-line separator is in shared/present, which is used by both jtk and cfl. This PR is scoped feat(jtk) but the render behavior change is global. No cfl tests are added to guard against silent regressions in cfl commands that emit consecutive DetailSections (e.g., page detail views). If cfl produces blank-line artifacts post-merge, there is no test contract to catch it.

**💡 Suggestion** - `shared/present/render_test.go:435`
> TestRender_DetailThenStderrMessage_NoStdoutSeparator has an inverted name: the test asserts wantStdout = "ID: 1\n\nID: 2\n", which verifies that a separator IS present between consecutive DetailSections. The name says 'NoStdoutSeparator' but the assertion confirms the opposite, which will mislead future readers.

</details>

<details>
<summary><strong>harness-engineering:enforcement-reviewer (2 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/present/emit.go:50`
> EmitIDs is variadic (ids ...string) while EmitIDsWithPagination takes a plain slice (ids []string). These are conceptually the same operation with asymmetric signatures. Call sites holding a slice must choose the pagination variant regardless of whether pagination is needed, and future migration of the remaining ~25 commands will encounter this inconsistency. Aligning both to the same convention would reduce friction.

**💡 Suggestion** - `tools/jtk/internal/present/comment.go:54`
> PresentListFull trims trailing newlines from the ADF body so the new consecutive-DetailSection separator produces exactly one blank line. PresentList (the non-fulltext/truncated path) is not visible in this diff. If PresentList also emits DetailSections with trailing newlines, those blocks will produce double blank lines between them under the new separator rule. Verify PresentList is table-mode only or apply the same trim.

</details>

<details>
<summary><strong>harness-engineering:knowledge-reviewer (2 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/cmd/issues/list.go:117`
> When len(result.Issues)==0 the text path calls Emit(PresentEmpty()), which carries no pagination context. If the API returns an empty page with hasMore=true, the continuation hint is silently dropped in text mode. The --id path avoids this because EmitIDsWithPagination is called first, but the text path does not apply the same appendPaginationHint logic.

**💡 Suggestion** - `tools/jtk/internal/cmd/comments/comments.go:80`
> Same empty-result + hasMore asymmetry as issues/list.go: when len(result.Comments)==0 the code returns PresentEmpty() via Emit, discarding the already-computed hasMore value. A mid-stream empty page in text mode silently drops the continuation hint.

</details>

<details>
<summary><strong>security:code-auditor (1 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/cmd/comments/comments.go:72`
> commentsHasMore produces a false positive when the final page is exactly maxResults in size and Total==0: the heuristic branch evaluates `got == maxResults` as true even when no further pages exist. After this PR this hint surfaces on stdout in both text and --id modes, making the false positive user-visible. Additionally, the degenerate case commentsHasMore(0, 0, 0, 0) returns true because `0 == 0`; a `got == 0 { return false }` guard would eliminate this edge.

</details>

*1 info-level observations excluded. Run with `--verbose` to include.*

*3 PR discussion threads considered.*

---
<details>
<summary>Completed in 4m 43s | $2.39</summary>

| Field | Value |
|---|---|
| Reviewers | hybrid-synthesis, database:sql-reviewer, security:code-auditor, harness-engineering:knowledge-reviewer, harness-engineering:enforcement-reviewer, harness-engineering:architecture-reviewer, harness-engineering:legibility-reviewer |
| Engine | claude |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 4m 43s wall · 13m 32s compute (Reviewers: 3m 14s · Synthesis: 1m 26s) |
| Cost | $2.39 |
| Tokens | 475.8k in / 34.7k out |
| Turns | 21 |

</details>

File: shared/present/render.go (hunk 1 above)
Comment (rianjs) [reply shared/present/render.go:11]: Declined — cfl has zero `DetailSection` usage today (`grep -r DetailSection tools/cfl` returns 0). The separator rule only activates when a model contains multiple stdout-bound DetailSections; with no consumers, there's no surface to silently regress. If cfl later adopts DetailSection in a multi-block render path, that adoption PR will see the spacing behavior in its own tests and can accept or adjust it then. Adding a cfl test for behavior cfl doesn't exercise is test-for-test's-sake.

File: tools/jtk/internal/cmd/comments/comments.go (hunk 2 above)
Comment (rianjs) [reply tools/jtk/internal/cmd/comments/comments.go:72]: Fixed in 2c344c76b29a1db80ff21154ae26a66c768adb3e — `commentsHasMore` now short-circuits to `false` when `got == 0`, eliminating the degenerate (0,0,0,0) case and protecting agents from infinite paging on empty responses where no cursor advances. The heuristic branch (Total=0, got==maxResults) is accepted as-is: without Total, we can't distinguish "exactly full page, done" from "full page, more to come" — erring toward false-positive means callers get one extra empty follow-up at worst. Added test cases: `empty page, total zero (no more)`, `empty page, total nonzero (no more)`, `degenerate all-zeros`.

File: tools/jtk/internal/cmd/issues/list.go (hunk 3 above)
Comment (rianjs) [reply tools/jtk/internal/cmd/issues/list.go:118]: Fixed in 2c344c76b29a1db80ff21154ae26a66c768adb3e — `issues list` empty-result path now routes through `AppendPaginationHint` so a mid-stream empty page with `IsLast=false` still surfaces the continuation line on stdout. Added `TestRunList_EmptyWithMoreResults_StillEmitsPaginationHint` to lock.

File: tools/jtk/internal/cmd/comments/comments.go (hunk 4 above)
Comment (rianjs) [reply tools/jtk/internal/cmd/comments/comments.go:80]: Addressed in 2c344c76b29a1db80ff21154ae26a66c768adb3e, but differently than you suggested: the security-auditor's `got == 0 → return false` guard on `commentsHasMore` already makes "empty page, more pages remain" impossible for comments (the heuristic would say hasMore=false because got=0, and with authoritative Total the agent has no cursor to advance on an empty page anyway). So no empty-path branch change is needed here — added `TestRunList_EmptyNeverEmitsSpuriousPaginationHint` to lock the interaction.

File: tools/jtk/internal/present/emit.go (hunk 5 above)
Comment (rianjs) [reply tools/jtk/internal/present/emit.go:56]: Fixed in 2c344c76b29a1db80ff21154ae26a66c768adb3e — `EmitIDs` is now `(opts, ids []string)` matching `EmitIDsWithPagination`. Call sites updated: `issues/get.go` wraps a single key as `[]string{issue.Key}`. Also exported `AppendPaginationHint` so command-layer callers can compose pagination onto any model (used by the new empty-path fix on `issues list`).

File: tools/jtk/internal/present/comment.go (hunk 6 above)
Comment (rianjs) [reply tools/jtk/internal/present/comment.go:54]: Declined — `PresentList` (the non-fulltext/truncated path at `comment.go:17`) emits a `TableSection`, not `DetailSection`. The renderer's blank-line separator rule only applies between consecutive `DetailSection`s, so truncated-mode comments cannot produce double blank lines regardless of ADF trailing-newline behavior. `PresentListFull` is the only DetailSection-emitting path and it already trims.

File: shared/present/render_test.go (hunk 7 above)
Comment (rianjs) [reply shared/present/render_test.go:435]: Fixed in 2c344c76b29a1db80ff21154ae26a66c768adb3e — renamed to `TestRender_StderrMessageBetweenDetails_StdoutSeparatorStillApplies` and added a comment explaining the invariant: the separator rule tracks the previous stdout-bound section, not the previous section overall, so an intervening stderr MessageSection does not break the chain.

File: shared/present/render.go (hunk 1 above)
Comment (monit-reviewer) [reply shared/present/render.go:11]: **Settled**: monit-reviewer flagged missing cfl tests for the global render behavior change. rianjs declined, noting cfl has zero DetailSection usage today so there's no surface to regress; future cfl adoption will cover it in that PR.

File: tools/jtk/internal/cmd/comments/comments.go (hunk 2 above)
Comment (monit-reviewer) [reply tools/jtk/internal/cmd/comments/comments.go:72]: **Settled**: monit-reviewer flagged a false positive in commentsHasMore for degenerate and exactly-full-page cases. rianjs fixed the degenerate case with a got==0 short-circuit and accepted the heuristic branch as-is, adding test cases to lock the behavior.

File: tools/jtk/internal/cmd/issues/list.go (hunk 3 above)
Comment (monit-reviewer) [reply tools/jtk/internal/cmd/issues/list.go:118]: **Settled**: monit-reviewer noted the text path dropped pagination hints on empty results. rianjs fixed it by routing the empty-result path through AppendPaginationHint and added a test to lock the behavior.

File: tools/jtk/internal/cmd/comments/comments.go (hunk 4 above)
Comment (monit-reviewer) [reply tools/jtk/internal/cmd/comments/comments.go:80]: **Settled**: monit-reviewer flagged the same empty-result pagination asymmetry in comments. rianjs addressed it differently — the got==0 guard on commentsHasMore makes the empty+more case impossible — and added a test locking the interaction.

File: tools/jtk/internal/present/emit.go (hunk 5 above)
Comment (monit-reviewer) [reply tools/jtk/internal/present/emit.go:56]: **Settled**: monit-reviewer flagged asymmetric signatures between EmitIDs (variadic) and EmitIDsWithPagination (slice). rianjs fixed it by aligning EmitIDs to take a slice, updated call sites, and exported AppendPaginationHint for composition.

File: tools/jtk/internal/present/comment.go (hunk 6 above)
Comment (monit-reviewer) [reply tools/jtk/internal/present/comment.go:54]: **Settled**: monit-reviewer asked to verify PresentList wouldn't produce double blank lines from trailing newlines. rianjs declined, noting PresentList emits a TableSection not DetailSection, so the separator rule doesn't apply.

File: shared/present/render_test.go (hunk 7 above)
Comment (monit-reviewer) [reply shared/present/render_test.go:435]: **Settled**: monit-reviewer flagged an inverted test name where the assertion contradicted the name. rianjs renamed the test and added an explanatory comment about the stdout-chain invariant.

File: tools/jtk/internal/cmd/issues/list.go [hunk 8]
```diff
@@ -107,26 +106,28 @@ func runList(ctx context.Context, opts *root.Options, project, sprint string, ma
...
 	}
 
+	hasMore := !result.Pagination.IsLast
+
+	if opts.EmitIDOnly() {
+		ids := make([]string, len(result.Issues))
+		for i, issue := range result.Issues {
+			ids[i] = issue.Key
+		}
+		return jtkpresent.EmitIDsWithPagination(opts, ids, hasMore)
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/issues/list.go:117]: 🔵 **Low** (harness-engineering:architecture-reviewer): Empty issues page with IsLast=false produces contradictory output: 'No issues found' and 'More results available' on stdout simultaneously. Unlike comments (which guards with got==0 in commentsHasMore), issues derives hasMore purely from !result.Pagination.IsLast with no empty-page guard. TestRunList_EmptyWithMoreResults_StillEmitsPaginationHint locks this in intentionally, but the two-message output is likely to confuse users and downstream parsers.

File: shared/present/render.go [hunk 9]
```diff
@@ -8,18 +8,27 @@ import (
...
+// by a blank line so block-mode lists (e.g. one DetailSection per comment)
+// render as distinct blocks rather than running together.
 func Render(model *OutputModel, style Style) RenderedOutput {
 	if model == nil {
 		return RenderedOutput{}
 	}
 	var stdout, stderr bytes.Buffer
+	var prevStdoutWasDetail bool
 	for _, section := range model.Sections {
 		text := renderSection(section, style)
```
Comment (monit-reviewer) [inline shared/present/render.go:22]: 🔵 **Low** (harness-engineering:architecture-reviewer): prevStdoutWasDetail is reset to false on any stdout-bound non-DetailSection (e.g. a StreamStdout MessageSection), so Detail → MessageSection(stdout) → Detail would suppress the blank-line separator between the two DetailSections. Coverage only validates the stderr-between-Details case; the stdout-MessageSection-between-Details case is untested and the behavior is undocumented. No current code triggers this, but it's a latent correctness gap as more commands migrate.

File: tools/jtk/internal/present/emit.go [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+
+// AppendPaginationHint returns sections with a pagination MessageSection
+// appended when hasMore is true, otherwise returns sections unchanged.
+// Every model-building pagination call site funnels through this so
+// wording, kind, and stream stay in sync across presenters and commands.
+func AppendPaginationHint(sections []present.Section, hasMore bool) []present.Section {
+	if !hasMore {
+		return sections
+	}
+	return append(sections, paginationMessageSection())
```
Comment (monit-reviewer) [inline tools/jtk/internal/present/emit.go:39]: 🔵 **Low** (harness-engineering:enforcement-reviewer): AppendPaginationHint uses append on the caller-supplied slice without documenting aliasing. All four current call sites pass freshly-constructed make([]Section, n) slices where len==cap, so there's no present bug. But since this helper is now exported and will be consumed by the ~25 pending command migrations, a future caller passing a resliced/pre-allocated slice with spare capacity could have data beyond len silently overwritten. A short contract note ('returns a new slice; may share backing array with input') would prevent foot-guns.

**Review by monit-reviewer — APPROVED** [review]
## Automated PR Review



**Reviewed commit:** `2c344c7` | **Previous:** `75eea54` (incremental)

> **Approved** with 3 non-blocking suggestions below. Address at your discretion.

### Summary

| Reviewer | Findings |
|----------|----------|
| harness-engineering:architecture-reviewer | 2 |
| harness-engineering:enforcement-reviewer | 1 |

<details>
<summary><strong>harness-engineering:architecture-reviewer (2 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/cmd/issues/list.go:116`
> Empty issues page with IsLast=false produces contradictory output: 'No issues found' and 'More results available' on stdout simultaneously. Unlike comments (which guards with got==0 in commentsHasMore), issues derives hasMore purely from !result.Pagination.IsLast with no empty-page guard. TestRunList_EmptyWithMoreResults_StillEmitsPaginationHint locks this in intentionally, but the two-message output is likely to confuse users and downstream parsers.

**💡 Suggestion** - `shared/present/render.go:22`
> prevStdoutWasDetail is reset to false on any stdout-bound non-DetailSection (e.g. a StreamStdout MessageSection), so Detail → MessageSection(stdout) → Detail would suppress the blank-line separator between the two DetailSections. Coverage only validates the stderr-between-Details case; the stdout-MessageSection-between-Details case is untested and the behavior is undocumented. No current code triggers this, but it's a latent correctness gap as more commands migrate.

</details>

<details>
<summary><strong>harness-engineering:enforcement-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/present/emit.go:34`
> AppendPaginationHint uses append on the caller-supplied slice without documenting aliasing. All four current call sites pass freshly-constructed make([]Section, n) slices where len==cap, so there's no present bug. But since this helper is now exported and will be consumed by the ~25 pending command migrations, a future caller passing a resliced/pre-allocated slice with spare capacity could have data beyond len silently overwritten. A short contract note ('returns a new slice; may share backing array with input') would prevent foot-guns.

</details>

*1 info-level observations excluded. Run with `--verbose` to include.*

*10 PR discussion threads considered.*

---
<details>
<summary>Completed in 5m 57s | $2.83</summary>

| Field | Value |
|---|---|
| Mode | Re-review · Cycle 2 · Session resumed |
| Reviewers | hybrid-synthesis, database:sql-reviewer, security:code-auditor, harness-engineering:knowledge-reviewer, harness-engineering:enforcement-reviewer, harness-engineering:architecture-reviewer, harness-engineering:legibility-reviewer |
| Engine | claude |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 5m 57s wall · 12m 56s compute (Reviewers: 3m 20s · Synthesis: 2m 12s) |
| Cost | $2.83 |
| Tokens | 865.8k in / 50.8k out |
| Turns | 23 |

</details>

File: tools/jtk/internal/cmd/issues/list.go (hunk 8 above)
Comment (rianjs) [reply tools/jtk/internal/cmd/issues/list.go:117]: Fixed in 721d88e15042cb0de3cbabd33c0bfe4ed34851a9 — the empty branch now splits by hasMore: false → `No issues found`, true → pagination hint only. The two-message output was my mistake in the previous commit; updated `TestRunList_EmptyWithMoreResults_EmitsOnlyPaginationHint` to lock the single-message contract.

File: shared/present/render.go (hunk 9 above)
Comment (rianjs) [reply shared/present/render.go:22]: Locked in 721d88e15042cb0de3cbabd33c0bfe4ed34851a9 — added `TestRender_StdoutMessageBetweenDetails_SuppressesSeparator` documenting that a stdout-bound MessageSection between two DetailSections resets the separator chain. Agreed no current code triggers this; the test makes the behavior intentional rather than accidental for future adopters.

File: tools/jtk/internal/present/emit.go (hunk 10 above)
Comment (rianjs) [reply tools/jtk/internal/present/emit.go:39]: Fixed in 721d88e15042cb0de3cbabd33c0bfe4ed34851a9 — added a docstring note on `AppendPaginationHint` about append aliasing semantics (may share backing array with input), so future callers passing pre-allocated slices with spare capacity are forewarned.
